apps/svelte.dev/src/routes/docs/[...path]/+page.svelte

```diff
@@ -17,6 +20,37 @@
 		const link = 'docs/' + data.document.file.split('/').slice(2).join('/');
 		return `https://github.com/sveltejs/${name}/edit/main/documentation/${link}`;
 	});
+
+	// make hash case-insensitive
+	// hash was lowercase in v4 docs and varying case in v5 docs
+	function get_url_to_redirect_to() {
+		const hash = $page.url.hash.replace(/^#/i, '');
+		if (!hash) return;
+
+		const elements = document.querySelectorAll('*[id]');
+		// if there's an exact match, use that. no need to redirect
+		// also handles the case where one appears twice with difference casing
+		// e.g. https://svelte.dev/docs/kit/@sveltejs-kit#redirect vs https://svelte.dev/docs/kit/@sveltejs-kit#Redirect
+		for (const el of elements) {
```

Comment:
why are we iterating? just use a case-insensitive selector

```js
const heading = document.querySelector(`[id="${id}" i]`);
```

Reply:
updated to use that, but we still need to iterate. e.g. we have https://svelte.dev/docs/kit/@sveltejs-kit#redirect and https://svelte.dev/docs/kit/@sveltejs-kit#Redirect

Reply:
could do this!

```js
document.querySelector(`[id="${id}"]`) ?? document.querySelector(`[id="${id}" i]`);
```

Reply:
well it turns out that this issue isn't fixable. we need to change one of these ids so that they don't collide: https://github.com/sveltejs/svelte.dev/issues/590